src/sdl2/video.rs

```diff
@@ -1177,23 +1177,13 @@ impl Window {
         }
     }
 
-    pub fn surface<'a>(&'a self, _e: &'a EventPump) -> Result<&'a SurfaceRef, String> {
+    pub fn surface<'a>(&'a mut self, _e: &'a EventPump) -> Result<Surface<'a>, String> {
```

Comment:
I agree that the `update_surface` method is currently pretty awkward. However, I would like to tackle that separately if at all possible. I think I would rather bring the `SurfaceRef` back for now and then we can together think of a good, elegant solution for this issue. I am currently thinking of having a whole separate struct instead of messing with the `SurfaceCanvas`. 

I have never used this method, so I am not 100% sure on hot it interacts with a renderer created from a window vs a surface. If I create it from a surface, and then get the `&mut Surface` using `surface_mut`, do I have to also call for `update_surface()`, or do I only have to do it if I get the surface using the `SDL_GetWindowSurface` method? Also, currently I can get and update the surface without having to create a Canvas (creating a renderer), so I am potentially forcing the user to use the SDL_Renderer. Given all of these (personal) unknowns, I would rather push it off for a later PR.